build/ci/postInstall.js

```
@@ -47,4 +47,29 @@ function fixJupyterLabDTSFiles() {
         console.log(colors.red(relativePath + ' file does not need updating.'));
     }
 }
+
+/**
+ * In order to get raw kernels working, we reuse the default kernel that jupyterlab ships.
+ * However it expects to be talking to a websocket which is serializing the messages to strings.
+ * Our raw kernel is not a web socket and needs to do its own serialization. To do so, we make a copy
+ * of the default kernel with the serialization stripped out. This is simpler than making a copy of the module
+ * at runtime.
+ */
+function createJupyterKernelWithoutSerialization() {
+    var relativePath = path.join('node_modules', '@jupyterlab', 'services', 'lib', 'kernel', 'default.js');
+    var filePath = path.join(constants_1.ExtensionRootDir, relativePath);
+    if (!fs.existsSync(filePath)) {
+        throw new Error("Jupyter lab default kernel not found '" + filePath + "' (pvsc post install script)");
+    }
+    var fileContents = fs.readFileSync(filePath, { encoding: 'utf8' });
+    var replacedContents = fileContents.replace(
+        /^const serialize =.*$/gm,
+        'const serialize = { serialize: (a) => a, deserialize: (a) => a };'
+    );
```

Please can we throw an exception if the required string is not replaced.
I.e. if the string has already been replaced, then ok, if not, then the code has changed, hence we should throw an exception at this point, else things will not work.

---

You mean check that the contents has changed?

---
In reply to: [416037369](https://github.com/microsoft/vscode-python/pull/11444#discussion_r416037369) [](ancestors = 416037369)